[ECODR-87] Update examples from monorepo

its-edmund opened the pull request.

its-edmund committed:
Update web examples

its-edmund committed:
Update node examples

its-edmund committed:
Update webpack examples

its-edmund committed:
Rename example folders

its-edmund committed:
Rename example folders

its-edmund committed:
Remove old examples

its-edmund committed:
Update example readmes

its-edmund committed:
Add in new examples

its-edmund committed:
Update folder names

its-edmund committed:
Delete old folders

its-edmund committed:
Merge branch 'update-examples-from-monorepo' into master-backup

its-edmund committed:
Fix merges again :(

its-edmund commented on Node/node-example-produce.js:
```diff
@@ -1,11 +1,3 @@
-=======
 import StreamrClient from 'streamr-client';
 
 // REPLACE THIS WITH YOUR OWN PRIVATE KEY
```
This now checks to see if you have replaced the private key with your own or not

its-edmund commented on Node/node-example-produce.js:
```diff
@@ -19,29 +11,20 @@ if (PRIVATE_KEY === 'YOUR-PRIVATE-KEY') {
-<<<<<<< HEAD
-    stream.publish(msg)
-=======
     client.publish(stream, msg)
```
Changed to updated method signatures

its-edmund commented on Node/package.json:
```diff
@@ -2,8 +2,19 @@
+  "scripts": {
+    "subscribe": "node node-example-subscribe.js",
+    "publish": "node node-example-produce.js"
+  },
```
Added easier npm scripts

its-edmund commented on Web/web-example-metamask.html:
```diff
@@ -1,6 +1,7 @@
+
 <head>
-    <script src="../../dist/streamr-client.web.js"></script>
+    <script src="https://unpkg.com/streamr-client@5.4.4/streamr-client.web.js"></script>
```
Replaced with cdn link

fonty1 replied:
Does unpkg support @ latest? I suspect this repo will be easier to maintain if it is always pulling the latest streamr clients

fonty1 replied:
@mxinburritos

its-edmund commented on Web/web-example-metamask.html:
```diff
@@ -46,14 +55,12 @@
-<script>
-    document.getElementById('get session token').addEventListener('click', getSessionToken)
-</script>
+    <button onclick='connectEthereum()'>Connect</button>
```
Added connect button for Metamask

its-edmund commented on Web/web-example-produce.html:
```diff
@@ -28,17 +44,17 @@
                 random: Math.random()
             }
 
+            if (!client) {
```
Checking for client in case user didn't replace stream id or private key to prevent an error from being thrown

its-edmund commented on Web/web-example-subscribe.html:
```diff
@@ -42,84 +44,85 @@
-    }
-
-    start()
-</script>
```
Nothing changed here, only spacing by linter

its-edmund commented on Webpack/.babelrc:
```diff
@@ -1,8 +1,4 @@
-  ],
-  "plugins": [
-    "add-module-exports"
-  ]
```
Had to add plugins to use async with webpack

its-edmund commented on Webpack/src/index.js:
```diff
@@ -45,14 +73,35 @@ document.getElementById('subscribe').addEventListener('click', () => {
         .then(() => log('Sent successfully: ', msg))
         .catch((err) => log(err))
+=======
+    client.getOrCreateStream({
```
Again, changed to use updated method signature

Commit:
- Rename 'produce' to 'publish'
- More info in the readme about what the examples will do
- Update streamr client to 'latest' in all instances

fonty1 reviewed (changes requested):
- @mxinburritos can you double check the Metamask and Webpack examples? Both did not work for me.

Metamask:
Needs to request access to get accounts from memory, a bit of googling will find the updated right way.

Webpack:
ERROR in ./src/index.js
Module build failed (from ./node_modules/babel-loader/lib/index.js):
SyntaxError: /Users/matthew/streamr/examples/Webpack/src/index.js: Unexpected token (9:0)

   7 | }
   8 | 
>  9 | <<<<<<< HEAD
     | ^

its-edmund commented:
Sorry for the delay, checking them now

its-edmund committed:
Fix unmerged files

its-edmund committed:
Update readme instructions

its-edmund commented:
Turns out, metamask needs a server running in order to work properly and it looks like there aren't any alternatives. I updated the readme for instructions about live-server.

fonty1 commented:
> Turns out, metamask needs a server running in order to work properly and it looks like there aren't any alternatives. I updated the readme for instructions about live-server.

Ahh, good catch.
